app/hooks/questions.ts

```diff
@@ -0,0 +1,138 @@
+import { BigNumber } from "ethers";
+import { useEffect, useReducer, useState } from "react";
+import { useContractWrite, useNetwork, usePrepareContractWrite } from "wagmi";
+import { iPFSdomain, questionStateEnum } from "~/utils/helpers";
+import type { IpfsData, QuestionData } from "~/utils/types";
+import { useBountyQuestionContract, useContracts, useQuestionStateControllerContract } from "./useContracts";
+
+export type getQuestionsByStateResult = {
+  questionId: BigNumber;
+  questionState: number;
+  totalVotes: BigNumber;
+  uri: string;
+  voters: string[];
+}[];
+
+export function useQuestions() {
+  const bountyQuestionContract = useBountyQuestionContract();
+  const questionStateControllerContract = useQuestionStateControllerContract();
+
+  const [questions, setQuestions] = useState<getQuestionsByStateResult>();
+
+  // Get chain data
+  useEffect(() => {
```

Comment:
This looks fine - so are we not using wagmi useContractRead hooks any longer but creating our own and just instantiating the contract first?

Reply:
It's just a different approach as I thought Wagmi didn't encourage that but instad to use useContractRead and pass that data down to a new component that uses that data - this works too just different architecture - This might make it possibly more scalable but also don't want to overcomplicate things and keep things as stupid as possible.

Reply:
The awkward part there is chaining those 2 `useContractRead` calls together. I think you have to use `enable` to disable one of the calls and then use `refetch` in a useEffect to chain them together. I thought this was a little nicer than that. Especially if in the future we do more chaining for reads.

Reply:
but the nice part is we can use this hook either way. The interface should remain the same. Only the internals change.